modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -358,6 +363,13 @@ def _get_new_qc(
             return new_query_compiler[new_query_compiler.columns[0]]
         if index_col is None:
             new_query_compiler._modin_frame.synchronize_labels(axis=0)
+        if not is_data_homogeneous:
+            ErrorMessage.missmatch_with_pandas(
+                operation="read_csv",
+                message="Data types of partitions are different! "
+                "Please refer to the troubleshooting section of the Modin documentation "
+                "to fix this issue",
+            )
```

Comment:
It is best to issue warnings as soon as possible.

Reply:
Yes, i agree. Warning was replaced.